rero_ils/modules/patron_transaction_events/api.py

```diff
@@ -149,18 +151,20 @@ def update_parent_patron_transaction(self):
         #   digits, we can multiply amounts by 100, cast result as integer,
         #   do operation with these values, and (at the end) divide the result
         #   by 100.
-        patron_transaction = self.patron_transaction()
-        total_amount = int(patron_transaction.get('total_amount') * 100)
-        if self.event_type == 'fee':
-            total_amount += int(self.amount * 100)
-        elif self.event_type in ('payment', 'cancel'):
-            total_amount -= int(self.amount * 100)
-        patron_transaction['total_amount'] = total_amount / 100
+        pttr = self.patron_transaction
```

Comment:
Is it possible to put it in an extension?

Reply:
not really. In some special cases we don't need to update the "parent pttr" when an event is created (it's piloted by an additional argument into `create` method). A `post_create` extension will always be triggered and we can't determine at this time if the parent will be update.